src/telemetrydeck-signal.test.tsx

```diff
@@ -0,0 +1,242 @@
+
+  /**
+   * This event listener is used to count the number of requests sent to the server.
+   */
+  let requestCount = 0;
+  server.events.on("request:start", () => {
+    requestCount += 1;
+  });
+  render(
+    <Setup td={td}>
+      <button
+        type='button'
+        onClick={async () => {
+          await mockQueue(td, "queue button click", { component, path });
+        }}
+      >
+        add to queue
+      </button>
+    </Setup>,
+  );
+  const addToQueue = screen.getByRole("button", { name: "add to queue" });
+  expect(requestCount).toBe(0);
+
+  await act(() => fireEvent.click(addToQueue));
+  expect(requestCount).toBe(0);
+});
+
+test("queued signals are sent together as one", async () => {
+  const td = createTelemetryDeck({ appID: "123456789", clientUser: "anonymous" });
+
+  /**
+   * This event listener is used to count the number of requests sent to the server.
+   * sorted by signal type and queue type respectively.
+   */
+  let signalSendRequestCount = 0;
+  let queueRequestCount = 0;
+  server.events.on("request:start", (request) => {
+    const body = request.body as { type: string }[];
+    body.forEach((signal: { type: string }) => {
+      if (signal.type === "signal button click") {
+        signalSendRequestCount += 1;
+      } else if (signal.type === "queue button click") {
+        queueRequestCount += 1;
+      }
+    });
+  });
+
+  render(
+    <Setup td={td}>
+      <>
+        <button
+          type='button'
+          onClick={async () => {
+            await mockSignal(td, "signal button click", { component, path });
+          }}
+        >
+          send signal
+        </button>
+        <button
+          type='button'
+          onClick={async () => {
+            await mockQueue(td, "queue button click", { component, path });
+          }}
+        >
+          add to queue
+        </button>
+        <button
+          type='button'
+          onClick={async () => {
+            await mockFlush(td);
+          }}
+        >
+          flush queue
+        </button>
+      </>
+    </Setup>,
+  );
+  const sendSignal = screen.getByRole("button", { name: "send signal" });
+  const addToQueue = screen.getByRole("button", { name: "add to queue" });
+  const flushQueue = screen.getByRole("button", { name: "flush queue" });
+  expect(signalSendRequestCount).toBe(0);
+  expect(queueRequestCount).toBe(0);
+
+  await act(() => fireEvent.click(sendSignal));
+  expect(signalSendRequestCount).toBe(1);
+  expect(queueRequestCount).toBe(0);
+
+  await act(() => fireEvent.click(addToQueue));
+  expect(signalSendRequestCount).toBe(1);
+  expect(queueRequestCount).toBe(0);
+
+  await act(() => fireEvent.click(addToQueue));
+  expect(signalSendRequestCount).toBe(1);
+  expect(queueRequestCount).toBe(0);
+
+  await act(() => fireEvent.click(flushQueue));
+  expect(signalSendRequestCount).toBe(1);
+  expect(queueRequestCount).toBe(2);
```

Comment:
These test cases are pretty simple, can you come up with more complex ones?

Reply:
I added a count for queued elements from the server response.